ionoscloud/resource_server.go

```diff
@@ -745,6 +751,57 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 			return diag.FromErr(utils.GenerateSetError("server", "inline_volume_ids", err))
 		}
 	}
+
+	if initialState, ok := d.GetOk("vm_state"); ok {
+		initialState := initialState.(string)
+		if initialState == "SHUTOFF" {
+			if serverType == "CUBE" {
+				diags := diag.FromErr(fmt.Errorf("cannot shut down a cube server, set to SUSPENDED instead"))
+				return diags
+			}
+			apiResponse, err := client.ServersApi.DatacentersServersStopPost(ctx, datacenterId, d.Id()).Execute()
```

Comment:
can we move this and below to a separate function? Create a server service similar to firewall and nic and move them there. The server functions are getting too large

Reply:
agree, for more complex operations which go beyond simply setting fields, as is the case of changing the power state of the server, it should be delegated to a service